compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4773,7 +4773,7 @@ private void checkFunctionInvocationExpr(BLangInvocation iExpr) {
 
     private void markAndRegisterClosureVariable(BSymbol symbol, DiagnosticPos pos) {
         BLangInvokableNode encInvokable = env.enclInvokable;
-        if (symbol.owner instanceof BPackageSymbol) {
+        if (symbol.owner instanceof BPackageSymbol && env.node.getKind() != NodeKind.ARROW_EXPR) {
```

Comment:
I'm guessing the issue here is that in `F1 f1 = a => b => a + b;`, `a`'s owner is the module?

Reply:
If that's the case, this should ideally be fixed by properly setting the owner of `a` to the symbol for the arrow function.

Reply:
If this is a blocker, I guess we can go with this for now and fix it properly later.

Reply:
I think the first check in this method is, we skip if the current symbol is a top level node. Is that correct for arrow functions which has closures?

Reply:
@pubudu91, isn't this related to the issue we discussed recently?

IIRC, even in the following example, the owner of `a` is not the arrow function - it is the function `foo`. Which again is not quite correct IMO.

```ballerina
function foo() {
    function (int) returns int x = a => a + 1;    
}
```

Reply:
@Kishanthan Yes, we skip the symbol if it's a module level construct (e.g., a module level variable) because those are handled in a different manner and don't need to deal with closures when referring to those symbols. 

In a normal function the owner of the param symbols is the function symbol. But in the arrow functions, looks like the relationship of the params aren't accurately reflected in the symbols. 

@MaryamZi Yeah, it seems to be the same issue

Reply:
So having a lambda module level function does not have this issue. I guess the reason here is that arrowFunction is **not** treated like lambda function at typechecking phase, or altest at closure analysis phase, whereas lambdas are specially/properly handled.

Then there is a bigger issue here IMO.